maas/resource_maas_device.go

```diff
@@ -135,12 +135,24 @@ func resourceDeviceUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 		Hostname:    d.Get("hostname").(string),
 		Zone:        d.Get("zone").(string),
 	}
-
 	device, err := client.Device.Update(d.Id(), &deviceParams)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	d.SetId(device.SystemID)
+	d.SetId(device.SystemID)	
+
+	if d.HasChange("network_interfaces") {
+		networkInterfaces := d.Get("network_interfaces").(*schema.Set).List()
```

Comment:
This one seems incomplete. You are missing the deletion of network interfaces that exist in MAAS but Terraform practitioner does not include in their plans and also the addition of network interfaces that are present in the plan and absent in MAAS. For both of the the `id` is not present in the state if I am not mistaken.